drivers/hmis/app/models/hmis/hud/custom_client_address.rb

```diff
@@ -21,10 +21,12 @@ class Hmis::Hud::CustomClientAddress < Hmis::Hud::Base
     :postal,
     :physical,
     :both,
+    MOVE_IN_TYPE = :move_in, # ag-specific
```

Comment:
While the address is "custom" in that it's a non-HUD field, it is not customer-specific. We expect it to be collected by several installations. "move in" is a specific HUD concept for permanent housing projects, they just don't specify a data structure for an address.

We need a separate column for this,  either:
1. New column `enrollment_address_type` that specifies _in what capacity is this address related to the EnrollmentID_. The value would be `move_in` but could incorporate other types in the future. (E.g. maybe address at exit, or prior address at intake etc... we don't have specific use-cases)
2. New boolean column `move_in_address`. Indicates whether this address is associated to the Enrollment.MoveInDate. 

I think I'd lean towards the more flexible approach of adding a new enum column. The only possible value for now would be null or `move_in`, for now.
And, I think there should be max 1 move-in address (uniqueness constraint on `data_source_id, EnrollmentID, PersonalID where enrollment_address_type=move_in and DateDeleted=null`). That would also let us easily resolve `Enrollment.moveInAddress` in GraphQL, which resolves my other comment re: how to choose which Address to resolve in the Move-In Form.


Thoughts? Happy to jump on a huddle if easier!

Reply:
Sounds good! I wasn't sure if we would need more than one.

> uniqueness constraint on data_source_id, EnrollmentID, PersonalID 

Do we need to include PersonalID on this constraint? Shouldn't (data_source_id, EnrollmentID) be enough?

Reply:
Yeah, it should be enough for HMIS installation environments, so maybe its OK.
Initially I included PersonalID because we are not actually there yet w.r.t. uniqueness of EnrollmentID (https://www.pivotaltracker.com/story/show/186232503/comments/238761730). It's not happening (anymore) in the HMIS installations, since we cleaned all that up from imported data. So its probably safe to leave out the personal id.

Reply:
Thanks for the explanation. I might leave this as an array that can only have one member in AR/graphql (enrollment.move_in_addresses). Existing code in Hmis::Hud::Processors::Base.construct_nested_attributes seems to expected a collection; it looks like it would require modification to support single-valued relations. Happy to do that if you think it's worth the time.